src/ruby/lib/grpc/generic/active_call.rb

```diff
@@ -205,14 +205,24 @@ def remote_send(req, marshalled = false)
     # list, mulitple metadata for its key are sent
     def send_status(code = OK, details = '', assert_finished = false,
                     metadata: {})
-      send_initial_metadata
+      metadata_already_sent = @send_initial_md_mutex.synchronize do
+        # Can't send Trailers-Only since Response-Headers have already been sent
+        v = @metadata_sent
+        @metadata_sent ||= true
+        v
+      end
+
       ops = {
         SEND_STATUS_FROM_SERVER => Struct::Status.new(code, details, metadata)
       }
       ops[RECV_CLOSE_ON_SERVER] = nil if assert_finished
+
+      # To send initial metadata as trailing metadata(Trailers-Only) in order to support retries.
+      # https://github.com/grpc/grpc/blob/92b45c8eefcdde5e0c0ec36373fdecb7cd7007d7/doc/PROTOCOL-HTTP2.md#responses
```

Comment:
This code change looks good. I'm thinking about this more though and I think there should be a ruby-level test for this, which would a) prevent us from ignoring this comment and breaking retries of ruby servers in the future, and b) provide an illustrative example of RPC retries from ruby actually happening

It would be great if we could add a ruby-equivalent of this [existing retry test](https://github.com/grpc/grpc/blob/master/test/core/end2end/tests/retry.cc#L100) that's written in C. Basically, the test would:

1) provide a "service config" channel arg to the client to configure retries for e.g. "aborted" status
2) send an RPC to the server
3) Accept the first RPC, and have the server return status "aborted", and check that the special "grpc-previous-rpc-attempts" header was zero at the server
4) Accept a second RPC at the server, and have the server return status "OK", and check that the special "grpc-previous-rpc-attempts" header from the client was 1.

I'd expect this test will be decently large. And in particular accepting RPC's in a certain order could be tricky; to deal with this, I suggest running the server with a "pool_size" parameter of 1. 

Let me know if you'd like help and I can try and send a patch to your branch